repose-aggregator/core/repose-core-api/src/main/scala/org/openrepose/core/filter/AbstractConfiguredFilter.scala

```diff
@@ -113,7 +113,14 @@ abstract class AbstractConfiguredFilter[T: ClassTag](val configurationService: C
     *
     * @return
     */
-  override def isInitialized: Boolean = initialized
+  override def isInitialized: Boolean = listenerInitialized
+
+  /**
+    * Indicates that the filter has done all initialization and is ready to serve.
+    *
+    * @return
+    */
+  def filterInitialized: Boolean = isInitialized
```

Comment:
Hm, so a way to differentiate between the listener in the abstract filter being initialized and the filter itself being initialized. Makes sense.

I don't love having to leave `isInitialized` public, though, since that might cause some confusion. I almost want to suggest having the abstract filter contain an `UpdateListener` rather than **be** an `UpdateListener`. That seems cleaner to me, I think...

Reply:
I thought the same, then i started to think out the implications of that change and it got pretty gross. I don't remember it all right now though.